marklogic-data-hub-spark-connector/src/test/java/com/marklogic/hub/spark/sql/sources/v2/writer/WriteDataWithOptionsTest.java

```diff
@@ -6,13 +6,102 @@
 import com.marklogic.hub.spark.sql.sources.v2.AbstractSparkConnectorTest;
 import org.junit.jupiter.api.Test;
 
+import org.apache.spark.sql.sources.v2.writer.WriterCommitMessage;
+
 import java.util.Set;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
 
 public class WriteDataWithOptionsTest extends AbstractSparkConnectorTest {
 
+    @Test
+    void testInvalidUriTemplate1() {
+        String uriTemplate = "{property1.json";
+        try {
```

Comment:
For consistency with other tests, do this:

```
RuntimeException ex = assertThrows(RuntimeException.class, () -> initializeDataWriter....);
assertEquals("...", ex.getMessage());
```

Reply:
Ready for commit